mgmt/LocalManager.cc

```diff
@@ -578,19 +578,18 @@ LocalManager::handleMgmtMsgFromProcesses(MgmtMessageHdr *mh)
     break;
   // Congestion Control - end
   case MGMT_SIGNAL_CONFIG_FILE_CHILD: {
-    static const MgmtMarshallType fields[] = {MGMT_MARSHALL_STRING, MGMT_MARSHALL_STRING, MGMT_MARSHALL_INT};
-    char *parent                           = nullptr;
-    char *child                            = nullptr;
-    MgmtMarshallInt options                = 0;
-    if (mgmt_message_parse(data_raw, mh->data_len, fields, countof(fields), &parent, &child, &options) != -1) {
-      configFiles->configFileChild(parent, child, (unsigned int)options);
+    MgmtMarshallInt options = 0;
+    MgmtMarshallString mparent;
+    MgmtMarshallString mchild;
+    if (mgmt_message_parse(data_raw, mh->data_len, &mparent, &mchild, &options) != -1) {
+      configFiles->configFileChild(mparent, mchild, (unsigned int)options);
     } else {
       mgmt_log("[LocalManager::handleMgmtMsgFromProcesses] "
                "MGMT_SIGNAL_CONFIG_FILE_CHILD mgmt_message_parse error\n");
     }
     // Output pointers are guaranteed to be NULL or valid.
-    ats_free_null(parent);
-    ats_free_null(child);
+    ats_free_null(mparent);
```

Comment:
Outside the scope of this PR, but `MgmtMarshallString` should probably be updated to use `ats_scoped_str` or `std::unique_ptr` internally.